### ISSUE TITLE
Automatic recompiling of dependencies

### ISSUE THREAD

**Issue by cpitclaudel** [raxod502/straight.el#295]
Hi there,

I tried to find the answer in the readme, but I didn't see it.  Sorry if I missed something.
How does straight.el handle recompiling dependencies?  In general, if package B depends on package A, and package A is updated, both A and B need to be recompiled. Otherwise, the compiled version of B will still contains the expanded form of macros originally supplied by A, leading to inconsistencies and bugs.

Does straight.el recompile dependencies after updating a package? (package.el doesn't)

Thanks!

**Comment by raxod502** [comment]
No, `straight.el` doesn't do this yet. It would be easy to add, and I would welcome a pull request adding that feature. I think it should suffice to do the following: when compiling a package, inspect the build cache (which contains dependency lists for all of its packages) and remove the entries for all packages that depend on the package being compiled. This would be done in the same place as the mtime is updated in the build cache after the package is compiled.

**Comment by raxod502** [comment]
See also #76, which deals with a related but different issue concerning correct byte-compilation of packages.

**Comment by cpitclaudel** [comment]
Thanks for the explanation and the pointer.   I haven't tried straight.el yet (I mostly use Cask).
